docs/design/src/mcmini/coordinator/coordinator.cpp

```diff
@@ -48,9 +79,24 @@ model_to_system_map::get_object_for_remote_process_handle(void *handle) const {
 model::state::objid_t model_to_system_map::record_new_object_association(
     void *remote_process_visible_object_handle,
     std::unique_ptr<model::visible_object_state> initial_state) {
-  // TODO: Create a new object through the coordinator and then map handle
-  // `remote_process_visible_object_handle` to the newly-created object.
-  return 0;
+  // Generate a new unique object ID. This could be based on a counter, UUID, or
+  // any other unique identifier strategy
+  static model::state::objid_t nextObjId =
+      0;  // Example using a simple counter, should be replaced with a
+          // thread-safe and more robust approach
+  model::state::objid_t newObjId = nextObjId++;
```

Comment:
The `model_to_system_map`  is just a way to restrict the API of the `coordinator`. Object ID assignment shouldn't be independent of the `_coordinator` the map wraps, since the underlying `model::program` object and this method must correspond. Instead rely on the `model::state` to vend you the ID via `add_object(...)`

Reply:
So essentially `_coordinator` --> `program` --> `state_sequence` and add the object there. Some extra methods or accessors I think might be needed